src/ios/GoogleMaps/PluginTotalPaveTileLayer.m

```diff
@@ -110,19 +110,19 @@ -(void)remove:(CDVInvokedUrlCommand *)command
     }];
 }
 
-- (void)pluginUnload {
-    NSArray *keys = [self.mapCtrl.objects allKeys];
-    NSString *key;
+- (void) pluginUnload {
+    NSArray* keys = [self.mapCtrl.objects allKeys];
     for (int i = 0; i < [keys count]; i++) {
-        key = [keys objectAtIndex:i];
+        NSString* key = [keys objectAtIndex:i];
         if ([key hasPrefix:PREFIX]) {
             key = [key stringByReplacingOccurrencesOfString:@"_property" withString:@""];
-            TotalPaveTileProvider *provider = (TotalPaveTileProvider *)[self.mapCtrl.objects objectForKey:key];
+            TotalPaveTileProvider* provider = (TotalPaveTileProvider*)[self.mapCtrl.objects objectForKey:key];
             provider.map = nil;
             provider = nil;
         }
+
+        [self.mapCtrl.objects removeObjectForKey:key];
```

Comment:
I'm under the impression that self.mapCtrl is shared between all plugins.

So if Polygon A adds GonA to objects and Polyline B adds LineB to objects.

When plugin unload happens, Polygon A will loop over objects, finding keys for GonA and LineB.
For all keys that has Polygon's prefix, it do additional clean up.

Then regardless if the key has the prefix or not, it will remove the object from mapCtrl.
So Polygons will remove LineB from mapCtrl.

Then Polylines won't get an opportunity to do their own additional cleanup, because mapCtrl.objects has already been cleared.


Can you confirm if this is true?

Reply:
Also, if there are multiple polygons, will polygons attempt to run this code multiple times?

Reply:
I'm following the existing pattern used by the other gmap plugins, e.g: https://github.com/totalpave/cordova-plugin-googlemaps/blob/master/src/ios/GoogleMaps/PluginCircle.m#L36

Reply:
> I'm following the existing pattern used by the other gmap plugins, e.g: https://github.com/totalpave/cordova-plugin-googlemaps/blob/master/src/ios/GoogleMaps/PluginCircle.m#L36

I know, but this looks like a problem as well. I guess I never clued into it when I was working with the totalpave tile layer. There's likely some reason why it hasn't caused errors yet but it would be better to understand the code instead of thinking it's really weird and hoping it doesn't cause some weird problem later.

Reply:
> Also, if there are multiple polygons, will polygons attempt to run this code multiple times?

No, I don't think so. Because these plugin objects are also just standard cordova plugins, there's only a single instance created and that instance acts as a logic controller for multiple objects. I don't see anywheres where the gmaps actually instantiates any of these plugin classes.

> I know, but this looks like a problem as well. I guess I never clued into it when I was working with the totalpave tile layer. There's likely some reason why it hasn't caused errors yet but it would be better to understand the code instead of thinking it's really weird and hoping it doesn't cause some weird problem later.

Maybe but that's out of scope and this PR isn't making it any worse, and we don't actually have any evidence of it causing any issues. I'm not going to change stuff on maybes and I don't have the time to refactor a good chunk of the plugin for a "maybe" issue.

The goal here is to address sentry issues that we **know** is an actual issue.